plane/src/drone/docker/commands.rs

```diff
@@ -187,3 +195,139 @@ pub async fn run_container(
+    async fn run_container_with_config(
+        config: bollard::container::Config<String>,
+    ) -> anyhow::Result<String> {
+        let docker = bollard::Docker::connect_with_local_defaults()?;
+        let image = config.image.clone().unwrap();
+        pull_image(&docker, &image, None, false).await?;
+        let backend_name = BackendName::new_random();
+        let container_id = ContainerId::from(format!("plane-test-{}", backend_name));
+        let create_container_options = bollard::container::CreateContainerOptions {
+            name: container_id.to_string(),
+            ..Default::default()
+        };
+
+        docker
+            .create_container(Some(create_container_options), config)
+            .await?;
+        docker
+            .start_container::<String>(&container_id.to_string(), None)
+            .await?;
+        let mut wait_container = docker.wait_container::<String>(&container_id.to_string(), None);
+
+        let mut output = String::new();
+        while let (Some(msg), Some(_)) = (
+            docker
+                .logs::<String>(
+                    &container_id.to_string(),
+                    Some(LogsOptions {
+                        follow: true,
+                        stdout: true,
+                        stderr: true,
+                        ..Default::default()
+                    }),
+                )
+                .try_next()
+                .await?,
+            wait_container.try_next().await?,
+        ) {
+            output = output + &msg.to_string();
+        }
+
+        docker
+            .remove_container(&container_id.to_string(), None)
+            .await?;
+
+        Ok(output)
+    }
+
+    #[tokio::test]
+    async fn test_resource_limits() -> anyhow::Result<()> {
+        let docker = bollard::Docker::connect_with_local_defaults()?;
+        let backend_name = BackendName::new_random();
+        let _container_id = ContainerId::from(format!("plane-test-{}", backend_name));
+        let mut executor_config = ExecutorConfig::from_image_with_defaults("debian:bookworm");
+
+        let resource_limits: ResourceLimits = serde_json::from_str(
+            r#"
+        {
+        "cpu_period": 1000000,
+        "cpu_time_limit": 100,
+        "cpu_period_percent": 80,
+        "memory_limit_bytes": 10000000,
+        "disk_limit_bytes": 10000000000
+        }
+        "#,
+        )?;
+
+        executor_config.resource_limits = resource_limits;
+        let info = docker.info().await?;
+        match info.driver {
+            Some(t) if matches!(t.as_str(), "btrfs" | "zfs") => {}
+            Some(t)
+                if t == "overlay2" && {
+                    let fs_ok;
+                    if let Some(status) = info.driver_status {
+                        fs_ok = !status.iter().flatten().any(|s| s.contains("xfs"));
+                    } else {
+                        fs_ok = false
+                    };
+                    !fs_ok
```

Comment:
I find the double-negation hard to file; consider omitting `fs_ok` entirely and implicitly returning the whole block:

```rust
if info.driver_status {
    status.iter().flatten().any(|s| s.contains("xfs"))
} else {
    true
}
```